llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -1184,15 +1184,15 @@ RecurrenceDescriptor::getReductionOpChain(PHINode *Phi, Loop *L) const {
   // more expensive than out-of-loop reductions, and need to be costed more
   // carefully.
   unsigned ExpectedUses = 1;
-  if (RedOp == Instruction::ICmp || RedOp == Instruction::FCmp)
+  if (isMinMaxRecurrenceKind(Kind))
```

Comment:
Instead of recalculating this each time, is it perhaps better to calculate it once and store it in a variable for reuse?